packages/matter.js/src/node/server/TransactionalInteractionServer.ts

```diff
@@ -119,18 +153,71 @@ export class TransactionalInteractionServer extends InteractionServer {
             session: exchange.session,
             tracer: this.#tracer,
             actionType: ActionTracer.ActionType.Read,
+            endpoint,
+            root: this.#endpoint,
         }).act(readAttribute);
     }
 
+    protected override async readEvent(
+        path: EventPath,
+        eventFilters: TypeFromSchema<typeof TlvEventFilter>[] | undefined,
+        event: EventServer<any, any>,
+        exchange: MessageExchange<MatterDevice>,
+        fabricFiltered: boolean,
+        message: Message,
+        endpoint: EndpointInterface,
+    ): Promise<EventStorageData<any>[]> {
+        const readEvent = (context: ActionContext) => {
+            // TODO Change this to behavior style, right now it does minimalistic whats needed
+            if (!context.hasAccessFor(event.readAcl, { cluster: path.clusterId } as AccessControl.Location)) {
+                throw new AccessDeniedError(
+                    `Access to ${endpoint.number}/${Diagnostic.hex(path.clusterId)} denied on ${exchange.session.name}.`,
+                );
+            }
+            return super.readEvent(path, eventFilters, event, exchange, fabricFiltered, message, endpoint);
+        };
+
+        return OnlineContext({
+            activity: (exchange as WithActivity)[activityKey],
+            fabricFiltered,
+            message,
+            session: exchange.session,
+            tracer: this.#tracer,
+            actionType: ActionTracer.ActionType.Read,
+            endpoint,
+            root: this.#endpoint,
+        }).act(readEvent);
+    }
+
+    override async handleWriteRequest(
+        exchange: MessageExchange<MatterDevice>,
+        writeRequest: WriteRequest,
+        message: Message,
+    ): Promise<WriteResponse> {
+        // TODO: This is a hack to prevent the ACL from updating while we are in the middle of a write transaction and will
+        //  be removed again once we somehow handle relevant sub transactions
+        this.aclServer.aclUpdateDelayed = true;
```

Comment:
This doesn't seem too bad given it's only necessary in very limited circumstances

Reply:
Problem is that also other will need that. Like groups cluster. Also here you have a list where we should process the final changes once at the end. Because here you start connecting to other nodes (or disconnecting). I think we need a more generic solution for that.

Reply:
I added a TODO

Reply:
Yes will need for groups too but I think groups & acls is it?  But can add subtransactions although I don't quite understand your first comment, maybe you could elaborate on the "why" in the project task.